plugin/EstateList.php

```diff
@@ -656,6 +661,86 @@ public function getEstateIds(): array
 		return array_column($this->_records, 'id');
 	}
 
+	/**
+	 * @param array $parkingArray
+	 * @param string $language
+	 * @param string $locate
+	 * @return array
+	 */
+	public function formatParkingLot(array $parkingArray, string $language, string $locate = 'de_DE'): array
+	{
+		$messages = [];
+		foreach ($parkingArray as $key => $parking) {
+			if (!$parking['Count']) {
+				continue;
+			}
+			switch ($language) {
+				case 'ENG':
+					$element = "{$parking['Count']} {$this->getParkingName($key, $parking['Count'])} at {$this->formatPrice($parking['Price'], $language, $locate)}";
+					break;
+				default:
+					$element = "{$parking['Count']} {$this->getParkingName($key, $parking['Count'])} à {$this->formatPrice($parking['Price'], $language, $locate)}";
+					break;
+			}
+			if (!empty($parking['MarketingType'])) {
+				$element .= " ({$parking['MarketingType']})";
+			}
+			array_push($messages, $element);
+
+		}
+		return $messages;
+	}
+
+	/**
+	 * @param string $str
+	 * @param string $language
+	 * @param string $locate
+	 * @return string
+	 */
+	public function formatPrice(string $str, string $language, string $locate): string
+	{
+		$digit = intval(substr(strrchr($str, "."), 1));
+		try {
+			$format = new NumberFormatter($locate, NumberFormatter::CURRENCY);
+			if ($digit) {
+				$format->setAttribute(NumberFormatter::MAX_FRACTION_DIGITS, 2);
+			} else {
+				$format->setAttribute(NumberFormatter::MAX_FRACTION_DIGITS, 0);
+			}
+			return str_replace("\xc2\xa0", " ", $format->formatCurrency($str, "EUR"));;
+		} catch (\Exception $exception){
+			if ($digit) {
+				$str = floatval($str);
+				$str = number_format_i18n($str, 2);
+			} else {
+				$str = number_format_i18n(intval($str));
+			}
+			switch ($language) {
+				case 'ENG':
+					$str = '€' . $str;
+					break;
+				default:
+					$str = $str . ' €';
+					break;
+			}
+			return $str;
+		}
+	}
+
+	/**
+	 * @param string $parkingName
+	 * @param int $count
+	 * @return string
+	 */
+	public function getParkingName(string $parkingName, int $count): string
```

Comment:
This highly depends on the language, WP has something for this, but the __* functions cannot be used with variables, but should always have a string literal as the text to be translated. See https://developer.wordpress.org/reference/functions/_n/ and https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#plurals